src/structure.cpp

```diff
@@ -619,7 +619,7 @@ structure_chunk::~structure_chunk() {
     delete[] chi2[c];
     delete[] chi3[c];
   }
-  FOR_DIRECTIONS(d) {
```

Comment:
This line can stay the same — only valid directions will ever be allocated.

Reply:
Fixed.